### ISSUE TITLE
Predict on out-of-training-set pipelines

### ISSUE THREAD

**Issue by bjschoenfeld** [byu-dml/d3m-dynamic-neural-architecture#30]
Some models cannot make predictions about pipelines which are not in the training set. (All pipelines must use primitives from the training set of course.) We need to show-case the models which can make predictions on novel pipelines.

To do this, the pipelines in the training data need to be restricted to some smaller subset. We should be careful that we do not eliminate any primitives from the training set. We would then have the models make predictions they can on the the full test set (with out-of-training-set pipelines). Those that can recommend pipelines from the full test set will likely have better scores, and vice-versa.

**Comment by erikhuck** [comment]
Does the train/test split not do this already? If not, is that what we should change? The train/test split. Or should we change the second train/set split that occurs at runtime.

**Comment by bjschoenfeld** [comment]
The train/test split happens across datasets. This would be another split, but across pipelines.
